src/main/java/ust/tad/analysismanager/analysistask/AnalysisTaskService.java

```diff
@@ -5,191 +5,207 @@
+    parentTask.addSubtask(subTask);
+
+    Process process = processes.get(parentTask.getTransformationProcessId());
+    process.activeTasks++;
+
+    return analysisTaskRepository.save(parentTask);
+  }
+
+  public AnalysisTask updateStatusFromAnalysisTaskResponse(AnalysisTaskResponse response) {
+    boolean responseSuccess = response.getSuccess();
+    UUID responseTaskId = response.getTaskId();
+
+    AnalysisTask task = analysisTaskRepository.getByTaskId(responseTaskId);
+
+    if (responseSuccess) {
+      task.setStatus(AnalysisStatus.FINISHED);
+    } else {
+      task.setStatus(AnalysisStatus.FAILED);
     }
 
-    public AnalysisTask updateStatusFromAnalysisTaskResponse(AnalysisTaskResponse response) {
-        boolean responseSuccess = response.getSuccess();
-        UUID responseTaskId = response.getTaskId();
-
-        AnalysisTask task = analysisTaskRepository.getByTaskId(responseTaskId);
-
-        if (responseSuccess) {
-            task.setStatus(AnalysisStatus.FINISHED);
-        } else {
-            task.setStatus(AnalysisStatus.FAILED);
-        }
-
-        Process process = processes.get(task.getTransformationProcessId());
-        process.activeTasks--;
-
-        if (process.visualize) {
-            if (process.activeTasks == 0 && process.visualizationTaskId == null && responseSuccess) {
-                AnalysisTask newVisualizationTask = new AnalysisTask();
-
-                newVisualizationTask.setTransformationProcessId(process.transformationProcessId);
-                newVisualizationTask.setTechnology("layout-pipeline");
-                newVisualizationTask.setCommands(process.commands);
-                newVisualizationTask.setLocations(null);
-                newVisualizationTask.setStatus(AnalysisStatus.WAITING);
-
-                AnalysisTask visualizationTask = analysisTaskRepository.save(newVisualizationTask);
-                addSubtask(process.mainTaskId, visualizationTask);
-                process.visualizationTaskId = visualizationTask.getTaskId();
-            } else if (process.activeTasks == 0 && responseTaskId.equals(process.visualizationTaskId)) {
-                processes.remove(process.transformationProcessId);
-            }
-        } else {
-            if (process.activeTasks == 0) {
-                processes.remove(process.transformationProcessId);
-            }
-        }
-        return analysisTaskRepository.save(task);
+    Process process = processes.get(task.getTransformationProcessId());
+    process.activeTasks--;
+
+    if (process.visualize) {
+      if (process.activeTasks == 0 && process.visualizationTaskId == null && responseSuccess) {
+        AnalysisTask newVisualizationTask = new AnalysisTask();
+
+        newVisualizationTask.setTransformationProcessId(process.transformationProcessId);
+        newVisualizationTask.setTechnology("layout-pipeline");
+        newVisualizationTask.setCommands(null);
```

Comment:
should this really be null?

Reply:
Since the commands are not used in the visualization, they can also be set to null here.